File: skill_sdk/tests/fixtures/ts-project/src/shop/discount/DiscountUtils.ts
import { DiscountStrategy } from './DiscountStrategy';
import { Order } from '../models/Order';

/**
 * Utility for applying multiple discount strategies.
 */
export class DiscountUtils {
  private constructor() {}

  /**
   * Tries several strategies and returns the lowest price.
   */
  static applyBestDiscount(order: Order, strategies: DiscountStrategy[]): number {
    let best = order.rawTotal();
    for (const s of strategies) {
      const value = s.apply(order);
      if (value < best) {
        best = value;
      }
    }
    return best;
  }
}

File: skill_sdk/tests/fixtures/java-project/src/main/java/com/fixture/shop/discount/DiscountUtils.java
package com.fixture.shop.discount;

import java.util.List;

import com.fixture.shop.models.Order;

/**
 * Utility for applying multiple discount strategies.
 */
public final class DiscountUtils {

    private DiscountUtils() {
    }

    /**
     * Tries several strategies and returns the lowest price.
     */
    public static double applyBestDiscount(Order order, List<DiscountStrategy> strategies) {
        double best = order.rawTotal();
        for (DiscountStrategy s : strategies) {
            double value = s.apply(order);
            if (value < best) {
                best = value;
            }
        }
        return best;
    }
}
